test/test_utils.py

```diff
@@ -310,6 +310,21 @@ def test_template(template: str, output: str) -> None:
             True,
             id="query_function_with_whitespace",
         ),
+        pytest.param(
+            "{{ some_filter(lookup('env', 'USER')) }}",
+            True,
+            id="nested_with_filter",
+        ),
+        pytest.param(
+            "{{ (query)('dict', my_var) }}",
+            True,
+            id="query_with_parentheses",
+        ),
+        pytest.param(
+            "{{ (q)('env', 'HOME') }}",
+            True,
+            id="q_with_parentheses",
+        ),
```

Comment:
Shouldn't there also be some examples where `has_lookup` is `False`? Like `{{ foo | lookup('env', 'HOME') }}` (here `lookup` is a filter and not the function), or something where `lookup` appears in a string?

Reply:
Added tests!